graphql-server/go-server/pkg/datasource/datasource.go

```diff
@@ -55,6 +55,7 @@ func datasource2model(obj *unstructured.Unstructured) *generated.Datasource {
 	updateTime := condition.LastTransitionTime.Time
 	status := string(condition.Status)
 	message := string(condition.Message)
+	common.TryToSetDeleting(&status, obj)
```

Comment:
```golang
func TryToSetDeleting(status string, obj *unstructured.Unstructured) string {
	if obj.GetDeletionTimestamp() != nil {
		return "Deleting"
	}
	return status
}

statu := common.TryToSetDelete(string(condition.Status),obj)

```

Reply:
If it is changed to a function with a return value, I think it can be done without the status parameter, and where this function is used it is first converted from unstructured to a concrete structure, so the function can be modified to a similar form:

```go
func GetObjectStatus(obj *unstructued.Unstructured) string {
   if obj.GetDeletionTimestamp() != nil {
     return "Deleting"
   }
   status := "Unknown"
   switch obj.GetKind() {
   case "Datasource":
      // do something
  }
  return status
}
```

Reply:
LGTM

Reply:
Be careful that for `Worker`, we use `Reason` as the status.